envoy/api/v2/route/route.proto

```diff
@@ -564,6 +564,15 @@ message RedirectAction {
 
   // [#not-implemented-hide:] The scheme portion of the URL will be swapped with "https".
   bool https_redirect = 4;
+
+  // [#not-implemented-hide:] Indicates that during redirection, the matched prefix (or path) should
+  // be swapped with this value. This option allows redirect URLs be dynamically created based on
+  // the request.
+  string prefix_rewrite = 5;
+
+  // [#not-implemented-hide:] Indicates that during redirection, the query portion of the URL will
+  // be removed. Works with *prefix_rewrite* and default value is false.
+  bool strip_query = 6;
```

Comment:
Do you think this will be useful outside redirect as well ? Like stripping query params for non redirects ? If so, we could have this option be replicated to the route as well. @htuch / @zuercher ??

Reply:
I can't think of an example where that would be useful that isn't contrived. I think we can leave it off for now.

Reply:
@rshriram Based on all our use-cases it was useful for redirects. Did not hit any specific case where this was needed during route forwarding.